sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5650,7 +5650,10 @@ static pi_result EventCreate(pi_context Context, pi_queue Queue,
 pi_result piEventCreate(pi_context Context, pi_event *RetEvent) {
   pi_result Result = EventCreate(Context, nullptr, true, RetEvent);
   (*RetEvent)->RefCountExternal++;
-  return Result;
+  if (Result != PI_SUCCESS)
+    return Result;
+  ZE_CALL(zeEventHostSignal, ((*RetEvent)->ZeEvent));
```

Comment:
We need to document new behavior in pi.h
And make sure all other backends follow the new semantics

Reply:
Other backends `die` with "non-implemented" as a reason.

Currently the function lacks documentation in the pi.h so we're not changing status quo in this regard by this change. All existing definitions of it are being updated by this PR.

Reply:
> Currently the function lacks documentation in the pi.h so we're not changing status quo in this regard by this change.

We are only documenting where PI definitions are diverging from similar OpenCL definitions (clCreateUserEvent in this case), and also bump up PI API version (at the top of pi.h)